### PR TITLE
Feature/pelagos 4785 add digitial resource type descriptor

### PR DESCRIPTION
**Pull request by mickel1138**

### CODE REVIEW

File: src/Entity/InformationProduct.php [hunk 1]
```diff
@@ -414,6 +428,46 @@ public function removeInformationProductType(InformationProductTypeDescriptor $i
...
+     * Add Digital Resource Type Descriptor to Information Product.
+     *
+     * @param DigitalResourceTypeDescriptor $digitalResourceTypeDescriptor
+     *
+     * @return self
+     */
+    public function addDigitalResourceTypeDescriptor(DigitalResourceTypeDescriptor $digitalResourceTypeDescriptor): self
+    {
+        if (!$this->digitalResourceTypeDescriptors->contains($digitalResourceTypeDescriptor)) {
+            $this->digitalResourceTypeDescriptors[] = $digitalResourceTypeDescriptor;
```
Comment (praneethpr) [inline src/Entity/InformationProduct.php:451]:
```suggestion
            $this->digitalResourceTypeDescriptors->add($digitalResourceTypeDescriptor);
```

File: src/Entity/InformationProduct.php [hunk 2]
```diff
@@ -414,6 +428,46 @@ public function removeInformationProductType(InformationProductTypeDescriptor $i
...
+     *
+     * @param DigitalResourceTypeDescriptor $digitalResourceTypeDescriptor
+     *
+     * @return self
+     */
+    public function removeDigitalResourceTypeDescriptor(DigitalResourceTypeDescriptor $digitalResourceTypeDescriptor): self
+    {
+        $this->digitalResourceTypeDescriptors->removeElement($digitalResourceTypeDescriptor);
+
+        return $this;
```
Comment (praneethpr) [inline src/Entity/InformationProduct.php:468]: Do we need the return on this?

File: src/Entity/InformationProduct.php [hunk 3]
```diff
@@ -414,6 +428,46 @@ public function removeInformationProductType(InformationProductTypeDescriptor $i
...
+     *
+     * @return self
+     */
+    public function addDigitalResourceTypeDescriptor(DigitalResourceTypeDescriptor $digitalResourceTypeDescriptor): self
+    {
+        if (!$this->digitalResourceTypeDescriptors->contains($digitalResourceTypeDescriptor)) {
+            $this->digitalResourceTypeDescriptors[] = $digitalResourceTypeDescriptor;
+        }
+
+        return $this;
```
Comment (praneethpr) [inline src/Entity/InformationProduct.php:454]: Do we need the return on this?

File: src/Event/InformationProductListener.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * Listener class for Information Product and Types related events.
+ */
+class InformationProductListener
+{
+    /**
+     * Listener on Pre-Remove event.
+     *
+     * @param LifecycleEventArgs $args The Lifecycle event arguments.
```
Comment (praneethpr) [inline src/Event/InformationProductListener.php:19]: Add throws tag

File: src/Entity/InformationProduct.php (hunk 3 above)
Comment (mickel1138) [reply src/Entity/InformationProduct.php:454]: It is added by make, I kinda like chaining, we should keep it.

File: src/Entity/InformationProduct.php (hunk 2 above)
Comment (mickel1138) [reply src/Entity/InformationProduct.php:468]: I like the chaining, so I think we should keep it.

**Review by fightingtexasaggie — APPROVED** [review]
looks good. If we end up making a bunch of these, should some of the more common crud functions be placed into a trait?

**Review by praneethpr — APPROVED** [review]
Looks good!

**Pull request merged**